backoffice/src/views/SingleBeaconRecordView.tsx

```diff
@@ -156,6 +202,15 @@ export const SingleBeaconRecordView: FunctionComponent<
         <TabPanel value={selectedTab} index={2}>
           <NotesPanel notesGateway={notesGateway} beaconId={beaconId} />
         </TabPanel>
+        <DialogueBox
+          isOpen={dialogueIsOpen}
+          dialogueTitle="Are you sure you want to permanently delete this record?"
+          dialogueContentText="This will delete the beacon record, its owner(s), and all other associated information."
+          action="Yes"
+          dismissal="No"
+          reasonsForAction={reasonsForDeletion}
```

Comment:
Is this the confirm dialog? If so, it shouldn't need the reasons?

Reply:
With React and Vue I've always worked in the school of thought of passing in state from the outside so child components are as close to stateless as poss. This is why I was passing in the reasons through DialogueBox to DeleteBeaconView. Do you prefer it if I do an API call to the new reasons endpoint from within DeleteBeaconView instead?

Reply:
Yeah that makes sense, but looks like this is the confirm dialog that doesn't need the reasons as you've already selected it in the previous dialog?

Reply:
Or is it the title of "Are you sure you want to permanently delete this record?" wrong and this is actually the reason selection dialog?

Reply:
This is the component structure:
- DialogueBox
      - Dialog
       - DeleteBeaconView
       - Rest of Dialog

DeleteBeaconView (first/reason selection stage of the dialogue) and the second/ confirm stage of the dialogue are siblings. DialogueBox is the parent that houses all of it and its props _dialogueTitle_ and _dialogueContentText_ are meant for the second/confirm stage. So DialogueBox passes in the reasons to DeleteBeaconView to populate the form inside DeleteBeaconView. Your question has deffo highlighted that I need to do some renaming here!

Reply:
Yeah, maybe passing in a "ConfirmModel" if ConfirmModel isn't null then add a confirm, and inside the model have the title/text for the confirm dialog?

Reply:
Maybe I should abstract this into a separate component something like ConfirmDialogue
`<div>
          <DialogTitle id="alert-dialog-title">{dialogueTitle}</DialogTitle>
          <DialogContent>
            <DialogContentText id="alert-dialog-description">
              {dialogueContentText}
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button onClick={() => handleClick(false)}>{dismissal}</Button>
            <Button onClick={() => handleClick(true)} autoFocus>
              {action}
            </Button>
          </DialogActions>
        </div>`